src/libraries/System.Net.Http/src/System/Net/Http/StreamContent.cs

```diff
@@ -97,11 +97,15 @@ protected override void Dispose(bool disposing)
             base.Dispose(disposing);
         }
 
-        protected override Stream CreateContentReadStream(CancellationToken cancellationToken) =>
-            new ReadOnlyStream(_content);
+        protected override Stream CreateContentReadStream(CancellationToken cancellationToken)
+        {
+            SeekToTheBeginningIfCan();
+            return new ReadOnlyStream(_content);
+        }
 
         protected override Task<Stream> CreateContentReadStreamAsync()
         {
+            SeekToTheBeginningIfCan();
```

Comment:
`PrepareContent()` here as well.

Reply:
The test MultipartContentTest.ReadAsStreamAsync_CanSeekEvenIfAllStreamsNotSeekale breaks because it does not except an exception when stream is non-seekable. I think the right thing here is to do just the seekable check without throwing an exception elsewhere